File: backend/internal/models/map.go
package models

import "time"

type PlayerMapSummary struct {
	SteamID    string `json:"steam_id"`
	BestTimeMS *int   `json:"best_time_ms"`
	Rank       *int   `json:"rank"`
	GroupTier  *int   `json:"group_tier"`
	Completed  bool   `json:"completed"`
}

type Map struct {
	ID                         int                `json:"id"`
	KSFMapID                   int                `json:"ksf_map_id"`
	Name                       string             `json:"name"`
	Tier                       int                `json:"tier"`
	Added                      int                `json:"added"`
	Completions                int                `json:"completions"`
	PlaytimeSeconds            int                `json:"playtime_seconds"`
	CompPerHour                float64            `json:"comp_per_hour"`
	Notes                      string             `json:"notes"`
	Bonus                      int                `json:"bonus"`
	Linear                     int                `json:"linear"`
	UpdatedAt                  time.Time          `json:"updated_at"`
	PlayerBestTimeMS           *int               `json:"player_best_time_ms"`
	PlayerRank                 *int               `json:"player_rank"`
	Completed                  *bool              `json:"completed"`
	PlayerRecords              []PlayerMapSummary `json:"player_records,omitempty"`
	RecordRefreshAvailableAt   *time.Time         `json:"record_refresh_available_at,omitempty"`
	RecordRefreshStatus        string             `json:"record_refresh_status,omitempty"`
	RecordRefreshFailedPlayers int                `json:"record_refresh_failed_players"`
}

File: frontend/src/types/Map.ts
export interface PlayerMapSummary {
    steam_id: string;
    best_time_ms?: number | null;
    rank?: number | null;
    group_tier?: number | null;
    completed: boolean;
}

export interface SurfMap {
    id: number;
    ksf_map_id: number;
    name: string;
    tier: number;
    added: number;
    completions: number;
    playtime_seconds: number;
    comp_per_hour: number;
    linear: number;
    notes: string;
    player_best_time_ms?: number | null;
    player_rank?: number | null;
    completed?: boolean | null;
    player_records?: PlayerMapSummary[];
}

export interface PaginatedMaps {
    maps: SurfMap[];
    total: number;
    page: number;
    per_page: number;
    total_pages: number;
}
